Sources/SwiftCrossUI/Environment/Environment.swift

```diff
@@ -36,21 +36,26 @@
 /// ```
 @propertyWrapper
 public struct Environment<Value>: DynamicProperty {
-    var keyPath: KeyPath<EnvironmentValues, Value>
+    var keyPath: KeyPath<EnvironmentValues, Value>?
+    var environmentKey: (any EnvironmentKey.Type)?
     var value: Box<Value?>
 
     public func update(
         with environment: EnvironmentValues,
         previousValue: Self?
     ) {
-        value.value = environment[keyPath: keyPath]
+        if let keyPath {
+            value.value = environment[keyPath: keyPath]
+        } else if let environmentKey {
+            value.value = (environment[environmentKey] as! Value)
```

Comment:
I've thought about how to avoid this force unwrap a little bit, and it seems like we can make `Value` a primary associated type of `EnvironmentKey`, and then update the environmentKey property to have type `any EnvironmentKey<Value>.Type`. This should make things a bit safer, and appears to be a backwards compatible change (because users of `EnvironmentKey` aren't forced to use the primary associated type syntax)?

I had to Google the syntax, so I figured I might as well include it here. To make `Value` a primary associatetype do the following;

```swift
protocol EnvironmentKey<Value> {
    associatedtype Value
    ...
}
```